### PR TITLE
Ported to go1.19, extra fixes

### PR DESCRIPTION
**Pull request by charbelcs123**
Ported to go1.19

Fixed few issues before building and running.

Was able to run using:
 oracall call -db-out ./pkg/db -pb-out ./pkg/pb -protoc-gen go -v -connect "user/pass@host:1521/DB"  'PKG.%'

### CODE REVIEW

File: custom/mmap_posix.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ func Mmap(f *os.File) ([]byte, error) {
 	}
 	p, err := syscall.Mmap(int(f.Fd()), 0, int(fi.Size()),
 		syscall.PROT_READ,
-		syscall.MAP_PRIVATE|syscall.MAP_DENYWRITE|syscall.MAP_POPULATE)
+		syscall.MAP_PRIVATE|0x800|0x8000)
```
Comment (tgulacsi) [inline custom/mmap_posix.go]: Why?

File: main.go [hunk 2]
```diff
@@ -262,16 +263,16 @@ func Main() error {
...
+				//cmd = exec.CommandContext(ctx,
+				//	"sed", "-i", "-e",
+				//	(`/timestamp "github.com\/golang\/protobuf\/ptypes\/timestamp"/ {s,timestamp.*$,timestamp "github.com/godror/knownpb/timestamppb",}; ` +
+				//		`/timestamppb "google.golang.org\/protobuf\/types\/known\/timestamppb"/ {s,timestamp.*$,timestamppb "github.com/godror/knownpb/timestamppb",}; `),
+				//	strings.TrimSuffix(pbFn, ".proto")+".pb.go",
+				//)
+				//cmd.Stdout, cmd.Stderr = os.Stdout, os.Stderr
+				//if err := cmd.Run(); err != nil {
+				//	return fmt.Errorf("%q: %w", cmd.Args, err)
+				//}
```
Comment (tgulacsi) [inline main.go]: Why?

File: main.go [hunk 3]
```diff
@@ -308,7 +309,7 @@ func Main() error {
 		},
 	}
 
-	fs = flag.NewFlagSet("oracall", flag.ContinueOnError)
+	//fs = flag.NewFlagSet("oracall", flag.ContinueOnError)
```
Comment (tgulacsi) [inline main.go]: Why?

**Review by tgulacsi — CHANGES_REQUESTED** [review]

File: main.go (hunk 3 above)
Comment (charbelcs123) [reply main.go]: Thought this is deprecated. Had no idea on how and why this should be used.
 Generated an error when testing it 
oracall oracall -v -connect "rator/rator@zh4labratordb2.sns.sec:1521/RATOR"       
                                                               
15:36:11.275 ERR "ERROR" attrs={"error":"terminal command (oracall) doesn't define an Exec function"}

File: main.go (hunk 2 above)
Comment (charbelcs123) [reply main.go]: Failing not sure why 

pkg/pb/pb.proto:5:1: warning: Import google/protobuf/timestamp.proto is unused.
sed: 1: "/timestamp "github.com\ ...": bad flag in substitute command: '}'
15:28:22.810 ERR "ERROR" attrs={"error":"[\"sed\" \"-i\" \"-e\" \"/timestamp \\\"github.com\\\\/golang\\\\/protobuf\\\\/ptypes\\\\/timestamp\\\"/ {s,timestamp.*$,timestamp \\\"github.com/godror/knownpb/timestamppb\\\",}; /timestamppb \\\"google.golang.org\\\\/protobuf\\\\/types\\\\/known\\\\/timestamppb\\\"/ {s,timestamp.*$,timestamppb \\\"github.com/godror/knownpb/timestamppb\\\",}; \" \"/Users/charbelsemaan/go/src/pkg/pb/pb.pb.go\"]: exit status 1"}

File: custom/mmap_posix.go (hunk 1 above)
Comment (charbelcs123) [reply custom/mmap_posix.go]: Got undefined when building on my mac (seems they are platform dependant)

custom/mmap_posix.go:33:31: undefined: syscall.MAP_DENYWRITE
custom/mmap_posix.go:33:53: undefined: syscall.MAP_POPULATE

Comment (tgulacsi) [reply custom/mmap_posix.go]: MAP_DENYWRITE is not what I thought of, MAP_POPULATE is just a slight optimization - just remove them!

File: main.go (hunk 2 above)
Comment (tgulacsi) [reply main.go]: It's better to fix those sed commands - remove the {} symbols! (or add a ; before the closing }, that's the real error - but for one command, {} grouping is not needed).

File: main.go (hunk 3 above)
Comment (tgulacsi) [reply main.go]: Each ffcli.Command needs a new FlagSet.
Invoke as

    oracall call -v -connect ...

**Pull request merged**